google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/SubscriberStats.java

```diff
@@ -16,65 +16,33 @@
 
 package com.google.cloud.pubsub;
 
-import com.google.common.math.Stats;
-import javax.annotation.concurrent.Immutable;
+import com.google.auto.value.AutoValue;
 
 /**
  * A snapshot of the subscriber statistics at the time they were requested from the {@link
  * Subscriber}.
  */
 //TODO: Finish implementation.
-@Immutable
-public class SubscriberStats {
-  private final long totalReceivedMessages;
-  private final long totalAckedMessages;
-  private final Stats endToEndLatency;
-  private final Stats ackLatency;
-  private final long numberOfAutoExtendedAckDeadlines;
-
-  SubscriberStats() {
-    this.totalReceivedMessages = 0;
-    this.totalAckedMessages = 0;
-    this.numberOfAutoExtendedAckDeadlines = 0;
-    this.endToEndLatency = null;
-    this.ackLatency = null;
-  }
-
-  /** Number of successfully published messages. */
-  public long getReceivedMessages() {
-    return totalReceivedMessages;
-  }
-
-  /** Number of successfully published messages. */
-  public long getAckedMessages() {
-    return totalAckedMessages;
-  }
+@AutoValue
+public abstract class SubscriberStats {
+  @AutoValue
+  public static abstract class Stats {}
```

Comment:
I'm not sure if I'm understanding you correctly, but...

No code references this yet, because the only way to get to Stats is through SubscriberStats, and the only place that returns SubscriberStats is not implemented (it just returns null for now).